Make a few cleanups and improvements to the STJ section of the 'What's New' article. (#42778)
dotnet/docs: docs/core/whats-new/dotnet-9/libraries.md

:::code language="csharp" source="../snippets/dotnet-9/csharp/Serialization.cs" id="RespectNullable":::

> > [!NOTE]

Author: Not sure if this markdown form is valid here, but didn't know how to best render a note section.

> > Due to how nullability annotations are represented in IL, the feature is restricted to annotations of non-generic properties. Please refer to the relevant documentation for more details.

Reviewer: Any guesses on what that "relevant documentation" might be?

Author: It's a placeholder for the conceptual documentation we need to be adding at some point. I have created a draft containing all the relevant content.

Reviewer 2: Should we comment that sentence out for now?

Author: Just the last sentence you mean? Sure.

:::code language="csharp" source="../snippets/dotnet-9/csharp/Serialization.cs" id="PropertyOrder":::

> ### Additional contract metadata APIs

Author: I removed this section since it didn't feel important enough to mention here.

> ### Customizing enum member names

Reviewer 2: ```suggestion
### Customize enum member names
```

The new <xref:System.Text.Json.Serialization.JsonStringEnumMemberNameAttribute?displayProperty=nameWithType> attribute can be used to customize the names of individual enum members for types that are serialized as strings:

```csharp
JsonSerializer.Serialize(MyEnum.Value1 | MyEnum.Value2); // "Value1, Custom enum value"

[Flags, JsonConverter(typeof(JsonStringEnumConverter))]
enum MyEnum
{
    Value1 = 1,
    [JsonStringEnumMemberName("Custom enum value")]
    Value2 = 2,
}
```

> ### Streaming multiple JSON documents

Reviewer 2: ```suggestion
### Stream multiple JSON documents
```

> <xref:System.Text.Json.Utf8JsonReader?displayProperty=nameWithType> now supports reading multiple, whitespace-separated JSON documents from a single buffer or stream. By default, it will throw an exception if it detects any non-whitespace characters that are trailing the first top-level document. This behavior can be changed using the <xref:System.Text.Json.JsonReaderOptions.AllowMultipleValues> flag:

Reviewer: Does this relate to JsonLines?

https://jsonlines.org/

Author: Essentially, although this particular implementation is more forgiving allowing arbitrary whitespace separation between values.

> <xref:System.Text.Json.Utf8JsonReader?displayProperty=nameWithType> now supports reading multiple, whitespace-separated JSON documents from a single buffer or stream. By default, it will throw an exception if it detects any non-whitespace characters that are trailing the first top-level document. This behavior can be changed using the <xref:System.Text.Json.JsonReaderOptions.AllowMultipleValues> flag:

Reviewer 2: ```suggestion
<xref:System.Text.Json.Utf8JsonReader?displayProperty=nameWithType> now supports reading multiple, whitespace-separated JSON documents from a single buffer or stream. By default, the reader throws an exception if it detects any non-whitespace characters that are trailing the first top-level document. You can change this behavior using the <xref:System.Text.Json.JsonReaderOptions.AllowMultipleValues> flag:
```

```csharp
JsonReaderOptions options = new() { AllowMultipleValues = true };
Utf8JsonReader reader = new("null {} 1 \r\n [1,2,3]"u8, options);

reader.Read();
Console.WriteLine(reader.TokenType); // Null

reader.Read();
Console.WriteLine(reader.TokenType); // StartObject
reader.Skip();

reader.Read();
Console.WriteLine(reader.TokenType); // Number

reader.Read();
Console.WriteLine(reader.TokenType); // StartArray
reader.Skip();

Console.WriteLine(reader.Read()); // False
```

> This additionally makes it possible to read JSON from payloads that may contain trailing data that is invalid JSON:

Reviewer 2: ```suggestion
This flag also makes it possible to read JSON from payloads that might contain trailing data that's invalid JSON:
```

```csharp
Utf8JsonReader reader = new("[1,2,3]    <NotJson/>"u8, new() { AllowMultipleValues = true });

reader.Read();
reader.Skip(); // Success
reader.Read(); // throws JsonReaderException
```

> When it comes to streaming deserialization, we have included a new <xref:System.Text.Json.JsonSerializer.DeserializeAsyncEnumerable> overload that makes streaming multiple top-level values possible. By default, the method will attempt to stream elements that are contained in a top-level JSON array. This behavior can be toggled using the new `topLevelValues` flag:

Reviewer 2: ```suggestion
When it comes to streaming deserialization, a new <xref:System.Text.Json.JsonSerializer.DeserializeAsyncEnumerable> overload makes streaming multiple top-level values possible. By default, the method attempts to stream elements that are contained in a top-level JSON array. You can toggle this behavior using the new `topLevelValues` flag:
```
